processing/src/main/java/io/druid/segment/incremental/OnheapIncrementalIndex.java

```diff
@@ -77,7 +77,7 @@
   {
     super(incrementalIndexSchema, deserializeComplexMetrics, reportParseExceptions, concurrentEventAdd);
     this.maxRowCount = maxRowCount;
-    this.maxBytesInMemory = maxBytesInMemory == 0 ? -1 : maxBytesInMemory;
+    this.maxBytesInMemory = maxBytesInMemory == 0 ? Runtime.getRuntime().maxMemory() / 6 : maxBytesInMemory;
```

Comment:
IMO, it's actually correct for this to be `-1`. It's because we expect the maxBytesInMemory to be passed in to the IncrementalIndex rather than set internally. A good sign of that is we don't have access to `TuningConfig` here.

Reply:
A bunch of test cases started failing, after I removed the check for `maxBytesInMemory < 0` due to OOM. So either we need to setMaxBytes, similar to `setRowCount` in all the tests, so it's not 0 or -1, which would lead to OOM. Or add back the check for `maxBytesInMemory <= 0`. The reason I removed that is, in `TuningConfigs`, it's set to `Long.MAX_VAL` if found -1. But we don't have access to `TuningConfig` here. Thoughts ?

Reply:
How about having the default here be `Long.MAX_VALUE` then? I was thinking just that it is correct for the IncrementalIndex to be unlimited as a default, but then to have a limit imposed on it in actual usage by callers (like Sink and such).

Reply:
Yes, that can work.